packages/core/src/lib/object_factory.js

```diff
@@ -22,14 +21,34 @@ const Pattern = function(relPath, data, patternlab) {
    * @param {patternlab} rendered html files for the pattern
    */
   const pathObj = path.parse(path.normalize(relPath));
+  const info = {};
+  // colors(.mustache) is subbed in 00-atoms-/00-global/00-colors
+  info.hasDir =
+    pathObj.dir.indexOf(pathObj.name) !== -1 ||
+    pathObj.dir.indexOf(pathObj.name.split('~')[0]) !== -1;
```

Comment:
The idea behind this was to keep the sorting in the folder and don't need it necessarily in the pattern part. But you are right. If I have something like atoms/buttons/button.* it will always think it is a nested Pattern, which is incorrect in this case.

Maybe your suggestion would be better and I change it to this.

```suggestion
    info.hasDir =
    path.basename(pathObj.dir) === pathObj.name ||
    path.basename(pathObj.dir) === pathObj.name.name.split('~')[0];
```